### ISSUE TITLE
[Feature Request] Ad-hoc netplay

### ISSUE THREAD

**Issue by rz5** [libretro/RetroArch#5092]
I felt we should have an issue for this, for further discussion.

On the IRC, I proposed that RetroArch have its netplay work like the PSP's ad-hoc WiFi feature. That is to say: for devices sufficiently close to each other, one device (sender, running RA) with a WiFi radio sends off a signal on some frequency/channel and one or more devices (listeners) are listening in on that frequency/channel.
The listeners, upon hearing the sender's signal, send off their acknowledge signals, handshakes occur, networking blackmagic and 2+ devices close to each other are now able to netplay together without the need of an extra router or access point.

**Comment by bparker06** [comment]
Two wifi devices can already netplay with each other without a router, they just need to join their own adhoc network that they create.

**Comment by rz5** [comment]
Right, that would be the best part - to be able to set this up within RA.
The way I'm imagining it is to have several funcs like adhoc_network_setup(), adhoc_host_create_session(), adhoc_guest_search_session() and inside each there would be a bunch of #ifdefs for platform specific stuff.

The most common use-case would be a pair of Android devices doing netplay; the most esoteric use-cases would be e.g. a Nintendo DS and a PS3 doing netplay.

**Comment by blackman91** [comment]
This would be cool on phones using wifi direct to use netplay without internet access when you are traveling.

**Comment by LibretroAdmin** [comment]
@Cthulhu-throwaway What do you think of this? Too ambitious/difficult?
